### PR TITLE
Index changes for listOperations, with refactor

### PR DESCRIPTION
**Pull request by werkt**
Returns index keys and group bindings for ""
Returns index keypairs with key match
Fixed logging on workers to emit pipeline messages with FINE
Reduced logging elsewhere to FINER for intra operation messages

### CODE REVIEW

File: _site/docs/execution/observability.md [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+An *Execution* is an individual Execute request made from a client to buildfarm. The names for all Executions on buildfarm will be the string `<instance name>/executions/<uuid>`, with `uuid` being generated after the Execute call is made.
+
+A *Tool Invocation* is an organization of all requests made by a client during one logical user invocation. For instance, `bazel`, as a _tool_, will generate a UUID per _invocation_, usually by a user or CI script - e.g: each time you run `bazel`, it generates internally and uses a different `toolInvocationId`, making this a collection of _Executions_ (and other requests) per run.
+
+A set of *Correlated Invocations* is uniquely identified by another UUID, the `correlatedInvocationsId`, which is expected to encompass one or more _Tool Invocations_. A build tool will provide the mechanism to specify this by a user - e.g: each time you run `bazel`, you can specify the same `--build_request_id` to indicate that a run is a member of a collection of _Tool Invocations_. The correlatedInvocationsId must be a string that _ends_ with an 8-4-4-4-12 format UUID string (important because this value is completely specified by a client user).
```
Comment (jasonschroeder-sfdc) [inline _site/docs/execution/observability.md:22]: consider documenting what happens if it _doesn't_ end in 8-4-4-4-12 UUID. Is it ignored? not correlated? Buildfarm crashes? 😂

File: _site/docs/execution/observability.md [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+## Indexing
+
+Note that the `correlatedInvocationsId` only needs to _end_ in a UUID. This means that before the end of the id, this can contain whatever identifying information we want.
+
+Buildfarm has taken steps that make it beneficial to provide this information as a URI. The `fragment` of the URI is an excellent place to specify the required trailing UUID. For example, our archetypal builder "Alice" provided this:
+
+```
+http://alice@dev-workstation/projects/winthorp/modules/interpreter?org=database&milestone=epsilon#fda159f2-bafe-4270-91ba-c7a946c0f2ba
```
Comment (jasonschroeder-sfdc) [inline _site/docs/execution/observability.md:31]:
```suggestion
https://alice@dev-workstation/projects/winthorp/modules/interpreter?org=database&milestone=epsilon#fda159f2-bafe-4270-91ba-c7a946c0f2ba
```

File: _site/docs/execution/observability.md [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+```
+
+She's specified a userinfo containing her `username`, the `host` the build ran on, a path that identified her `project` and its `module` per her organization, which is listed here as `database`, and she's clearly working on the 'epsilon' `milestone`. Trailing it after a fragment separator is the required uuid.
+
+A single layer of optional indexing is provided in buildfarm to collect _Correlated Invocations_ with useful identifying information.
+
+The indexing starts with the configuration of buildfarm's server:
+
+```
```
Comment (jasonschroeder-sfdc) [inline _site/docs/execution/observability.md:40]:
```suggestion
```yaml
```

File: _site/docs/execution/observability.md [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+
+Note that the `correlatedInvocationsId` only needs to _end_ in a UUID. This means that before the end of the id, this can contain whatever identifying information we want.
+
+Buildfarm has taken steps that make it beneficial to provide this information as a URI. The `fragment` of the URI is an excellent place to specify the required trailing UUID. For example, our archetypal builder "Alice" provided this:
+
+```
+http://alice@dev-workstation/projects/winthorp/modules/interpreter?org=database&milestone=epsilon#fda159f2-bafe-4270-91ba-c7a946c0f2ba
+```
+
+She's specified a userinfo containing her `username`, the `host` the build ran on, a path that identified her `project` and its `module` per her organization, which is listed here as `database`, and she's clearly working on the 'epsilon' `milestone`. Trailing it after a fragment separator is the required uuid.
```
Comment (jasonschroeder-sfdc) [inline _site/docs/execution/observability.md:34]: Spell-out that passwords should not be sent and will be ignored?

File: src/main/java/build/buildfarm/instance/InstanceBase.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+  @Override
+  public String getName() {
+    return name;
+  }
+
+  private String bind(String binding, UUID id) {
+    String name = getName() + "/" + binding;
+    if (name.isEmpty()) {
+      return name;
+    }
```
Comment (jasonschroeder-sfdc) [inline src/main/java/build/buildfarm/instance/InstanceBase.java:25]: bug: name can never be empty as it's will have a "/" in it

File: _site/docs/execution/observability.md (hunk 2 above)
Comment (werkt) [reply _site/docs/execution/observability.md:31]: Functionally this scheme is meaningless, but I'll swap it.

File: src/main/java/build/buildfarm/instance/InstanceBase.java (hunk 5 above)
Comment (werkt) [reply src/main/java/build/buildfarm/instance/InstanceBase.java:25]: I really have no idea what I was thinking with this - probably got stuck in there from an earlier version of string mutation collapse. I'll just return the single line concat here.

**Pull request merged**